nautobot/core/models/__init__.py

```diff
@@ -50,3 +97,92 @@ def validated_save(self):
         """
         self.full_clean()
         self.save()
+
+    def natural_key(self):
+        """
+        Smarter default implementation of natural key construction.
+
+        1. Handles nullable foreign keys (https://github.com/wq/django-natural-keys/issues/18)
+        2. Handles variadic natural-keys (e.g. Location model - [name, parent__name, parent__parent__name, ...].)
+        """
+        vals = []
+        for name_list in [name.split("__") for name in self.get_natural_key_fields()]:
+            val = self
+            for attr_name in name_list:
+                val = getattr(val, attr_name)
+                if val is None:
+                    break
+            vals.append(val)
+        # Strip trailing Nones from vals
```

Comment:
This means an (as-yet-undocumented) best practice is, for any model like Location that has a variadic natural key, the variadic portion *needs* to be the last item in the natural key - i.e. `(name, parent)`, not `(parent, name)` in the Location case.

Reply:
Variadic in that it depends on how deeply nested it is?

Reply:
Yeah exactly - it can be `("me")` (if no parent), or `("me", "parent", "grandparent")`, or `("me", "parent", "grandparent", "great-gp", ...)`.